crates/ironrdp-rdpdr/src/lib.rs

```diff
@@ -56,11 +59,42 @@ impl Rdpdr {
 
     #[must_use]
     pub fn with_smartcard(mut self, device_id: u32) -> Self {
-        self.device_list.add_smartcard(device_id);
         self.capabilities.add_smartcard();
+        self.device_list.add_smartcard(device_id);
```

Comment:
Question: if the order is especially important, do you think you could add a comment explaining why? (seems easy to break otherwise)

Reply:
It's not actually important, I just swapped it because I used the other ordering in `with_drives`.